pkg/gui/style/text_style.go

```diff
@@ -0,0 +1,160 @@
+package style
+
+import (
+	"github.com/gookit/color"
+)
+
+// A TextStyle contains a foreground color, background color, and
+// decorations (bold/underline/reverse).
+//
+// Colors may each be either 16-bit or 256-bit RGB colors. When
+// we need to produce a string with a TextStyle, if either foreground or
+// background color is RGB, we'll promote the other color component to RGB as well.
+// We could simplify this code by forcing everything to be RGB, but we're not
+// sure how compatible or efficient that would be with various terminals.
+// Lazygit will typically stick to 16-bit colors, but users may configure RGB colors.
+//
+// TextStyles are value objects, not entities, so for example if you want to
+// add the bold decoration to a TextStyle, we'll create a new TextStyle with
+// that decoration applied.
+//
+// Decorations are additive, so when we merge two TextStyles, if either is bold
+// then the resulting style will also be bold.
+//
+// So that we aren't rederiving the underlying style each time we want to print
+// a string, we derive it when a new TextStyle is created and store it in the
+// `style` field.
+
+type TextStyle struct {
+	fg         *Color
+	bg         *Color
+	decoration Decoration
+
+	style Sprinter
+}
+
+type Sprinter interface {
+	Sprint(a ...interface{}) string
+	Sprintf(format string, a ...interface{}) string
+}
+
+func New() TextStyle {
+	s := TextStyle{}
+	s.style = s.deriveStyle()
+	return s
+}
+
+func FromBasicFg(fg color.Color) TextStyle {
+	s := New()
+	c := NewBasicColor(fg)
+	s.fg = &c
```

Comment:
turns out you need to assign a value to a variable before you can take a pointer of it which is a bit weird but only requires one extra line